File: src/main/java/com/instagram/api/controller/UserController.java
package com.instagram.api.controller;

import com.instagram.api.common.response.CommonResponse;
import com.instagram.api.dto.UserJoinReqDto;
import com.instagram.api.service.UserService;
import lombok.RequiredArgsConstructor;
import org.springframework.http.ResponseEntity;
import org.springframework.web.bind.annotation.PostMapping;
import org.springframework.web.bind.annotation.RequestBody;
import org.springframework.web.bind.annotation.RequestMapping;
import org.springframework.web.bind.annotation.RestController;

@RequiredArgsConstructor
@RestController
@RequestMapping("/user")
public class UserController {
    private final CommonResponse commonResponse;
    private final UserService userService;

    /**
     * 회원가입
     * @param userJoinReqDto
     * @return
     */
    @PostMapping("join")
    public ResponseEntity<?> addUser(@RequestBody UserJoinReqDto userJoinReqDto) {
        userService.addUser(userJoinReqDto);
        return commonResponse.send();
    }
}

File: kotlin-app_v2/src/main/kotlin/com/group/libraryapp/controller/user/UserController.kt
package com.group.libraryapp.controller.user

import com.group.libraryapp.dto.user.request.UserCreateRequest
import com.group.libraryapp.dto.user.request.UserUpdateRequest
import com.group.libraryapp.dto.user.response.UserLoanHistoryResponse
import com.group.libraryapp.dto.user.response.UserResponse
import com.group.libraryapp.service.user.UserService
import org.springframework.web.bind.annotation.*

@RestController
class UserController(
    private val userService: UserService
) {
    @PostMapping("/user")
    fun saveUser(@RequestBody request: UserCreateRequest) {
        userService.saveUser(request)
    }

//    @GetMapping("/user")
//    fun getUsers(): List<UserResponse> = userService.getUsers()

    @GetMapping("/user")
    fun getUsers(): List<UserResponse> {
        return userService.getUsers()
    }

    @PutMapping("/user")
    fun updateUserName(@RequestBody request: UserUpdateRequest) {
        userService.updateUserName(request)
    }

    @DeleteMapping("/user")
    fun deleteUser(@RequestParam name: String) { // @RequestParam 는 default 가 required true (String? 가 되면 required false로 한다)
        userService.deleteUser(name)
    }

    @GetMapping("/user/loan")
    fun findAllWithHistoryList(): List<UserLoanHistoryResponse> {
        return userService.getUserLoanHistories()
    }
}